grafana/grafana
# InfluxDB: Move database information into jsondata

**What is this feature?**
Since the `database` field in `DataSourceInstanceSettings` is deprecated we have to move that information in to the `jsonData`.
Newly created InfluxDB v1.x datasources will have `dbName` field in `jsonData` to store database information. 
Existing datasources will not be affected and will keep working. If the user updates the datasource configuration (changes the database name) new data will be written in jsonData. 

Also, test data for local development is updated and they do not have `database` field but `jsonData.dbName`.

**Why do we need this feature?**

`database` field in `DataSourceInstanceSettings` is deprecated. https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/datasource.ts#L583-L587
We should move that information into the `jsonData`.

**Who is this feature for?**

People who use InfluxDB v1.x

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/59115

**Special notes for your reviewer**:

How to test
- Create an influxdb v1.x datasource/dashboard or use an existing one. 
- Switch to this branch and see datasource and queries and dashboards are working.

Disclaimer
- A broader deprecation and backward compatibility I will have another task. This won't be easy and we should be careful about that.
